### DISCUSSION TITLE
Is incrementing thread pool count possible?

### DISCUSSION THREAD

**Discussion by giacomodeliberali** [quartznet/quartznet#1070] [Q&A] [answered]
Is is possible at runtime to increment or decrement the number of thread in the pool?

**Comment by lahma** [comment, marked as answer]
This is not supported. You would need to implement your own `IThreadPool` implementation. The default one uses task scheduler's pool and when configured to have high enough value for `MaxConcurrency` should server most purposes. I think making it change during runtime might cause unanticipated behavior. High enough value should utilize server resources in optimal way.

What's your use case for making this dynamic?

**Reply by giacomodeliberali** [reply]
I was wondering the implications of directly allocating a high enough number of threads vs allocating a small amount and scale when needed. But I do suspect that even on small load having a bunch of threads waiting is not a big deal

**Reply by lahma** [reply]
And when you use the defaults, it actually isn't creating any threads, it's using CLR managed thread pool. You only get Quartz-generated threads if you choose to go with `DedicatedThreadPool`. So with defaults you are effectively using the very same thread pool as your async Tasks.
